src/lib.rs

```diff
@@ -65,12 +67,54 @@ impl OfferHandler {
     }
 
     /// Adds an offer to be paid with the amount specified. May only be called once for a single offer.
-    pub fn pay_offer(&mut self, _offer: Offer, _amount: u64) -> Result<(), OfferError> {
-        /*
-           Check if we've already added offer -> error.
-           Add offer to state map.
-           Create invoice request and push to offer queue.
-        */
+    pub async fn pay_offer(
+        &self,
+        offer: Offer,
+        amount: Option<u64>,
+        network: Network,
+        client: Client,
+        blinded_path: BlindedPath,
+    ) -> Result<(), OfferError<Secp256k1Error>> {
+        sleep(Duration::from_secs(10)).await;
```

Comment:
This sleep is not ideal... But we need the offer handler to be up and running before we pay the offer, otherwise racey things happen. One possible way of handling this would be to feed the OfferHandler a channel that pay_offer looks at. Pay_offer waits to start paying until it gets an update from this channel. Thoughts?